### ISSUE TITLE
Enhanced Filter Documentation 📚

### ISSUE THREAD

**Issue by GrantBirki** [nobe4/gh-not#86]
# Filter Documentation

This issue requests _enhanced_ filter documentation. Perhaps its just me being 🥔 🧠 but I am struggling to figure out how to get the following filter working:

```yaml
- name: ignore noisy review requests
    action: done
    filters:
      - (.repository.full_name == "org/repo1" or .repository.full_name == "org/repo2")
      - .reason == "review_requested"
```

I have tried a few variations of this filter:

- Attempt 1:
  
  ```yaml
  (.repository.full_name == "org/repo1" or .repository.full_name == "org/repo2")
  ```

- Attempt 2:

  ```yaml
  filters:
      - >
        (.repository.full_name == "org/repo1") or
        (.repository.full_name == "org/repo2") and
        .reason == "review_requested"
  ```

I have also tested `review-requested` instead of `review_requested` without luck as well.

Here is what I'm trying to get my filter to do in plain English:

> _Mark notifications as **done** that are from `org/repo1`, or `org/repo2` repos with the `review_requested` status_.

**Comment by nobe4** [comment]
Hi @GrantBirki, thanks for your patience! I am just coming to that.

In https://github.com/nobe4/gh-not/pull/87 I decided to implement a comprehensive list of tests that also showcase how filter can work. This might serve as a documentation for the time being.

As for your case, the first filter that you wrote is supposed to work, see 0232ef5.

Can you check that your notification cache contains a notification for this case? Maybe with something like

```shell
cat path/to/cache.json | jq '.[] | select(.repository.full_name == "org/repo1" or .repository.full_name == "org/repo2") | .reason' | sort -u
```

E.g. result for my cache:
```shell
"review_requested"
"state_change"
```

**Comment by nobe4** [comment]
Interesting, what happens if you run
```shell
$ cat ~/.local/state/gh-not/cache.json | jq '.[].reason' | sort -u
```

Here's what I have:
```
"assign"
"author"
"ci_activity"
"comment"
"mention"
"review_requested"
"state_change"
"subscribed"
"team_mention"
```

`state_change` is in the list, as are others reasons.

**Comment by GrantBirki** [comment]
I think this can be closed out as it all works for me now. It turned out that I had some syntax issues in my yaml config.
